server/GraknServer.java

```diff
@@ -147,52 +151,64 @@ private static Properties parseProperties() {
+        final CommandLine commandLine = new CommandLine(startCommand).addSubcommand(importDataCommand);
+        commandLine.setDefaultValueProvider(new PropertiesDefaultProvider(properties));
 
         try {
-            command.parseArgs(args);
-            if (command.isUsageHelpRequested()) {
-                command.usage(command.getOut());
-                proceed = false;
-            } else if (command.isVersionHelpRequested()) {
-                command.printVersionHelp(command.getOut());
-                proceed = false;
+            CommandLine.ParseResult parseResult = commandLine.parseArgs(args);
+            if (commandLine.isUsageHelpRequested()) {
+                commandLine.usage(commandLine.getOut());
+                return null;
+            } else if (commandLine.isVersionHelpRequested()) {
+                commandLine.printVersionHelp(commandLine.getOut());
+                return null;
             } else {
-                proceed = true;
+                if (parseResult.hasSubcommand()) {
+                    assert parseResult.subcommand().asCommandLineList().size() == 1;
+                    return parseResult.subcommand().asCommandLineList().get(0).getCommand();
+                } else {
+                    assert parseResult.asCommandLineList().size() == 1;
+                    return parseResult.asCommandLineList().get(0).getCommand();
+                }
             }
         } catch (ParameterException ex) {
-            command.getErr().println(ex.getMessage());
-            if (!UnmatchedArgumentException.printSuggestions(ex, command.getErr())) {
-                ex.getCommandLine().usage(command.getErr());
+            commandLine.getErr().println(ex.getMessage());
+            if (!UnmatchedArgumentException.printSuggestions(ex, commandLine.getErr())) {
+                ex.getCommandLine().usage(commandLine.getErr());
             }
-            proceed = false;
+            return null;
         }
-
-        if (proceed) return options;
-        else return null;
     }
 
     public static void main(String[] args) {
         try {
             final long start = System.nanoTime();
 
             printASCIILogo();
-            final ServerOptions options = parseCommandLine(parseProperties(), args);
-            if (options == null) System.exit(0);
-
-            final GraknServer server = new GraknServer(options);
-            server.start();
-
-            final long end = System.nanoTime();
-            LOG.info("Grakn Core version: {}", Version.VERSION);
-            LOG.info("Grakn Core Server has been started (in {} ms)",
-                     String.format("%.3f", (end - start) / 1_000_000.00));
-
-            server.serve();
+            final ServerCommand command = parseCommandLine(parseProperties(), args);
+            if (command == null) System.exit(0);
+
+            if (command instanceof ServerCommand.Start) {
```

Comment:
our pattern would be to implement `ServerCommand.isStart()` in the interface as a default boolen returning false, then overriding it in each subclass once